### PR TITLE
Adding new GPS conversion convenience function.

### PR DESCRIPTION
**Pull request by enricgalceran**

### CODE REVIEW

**Comment by tcies** [comment]
add to whitelist

File: msf_core/include/msf_core/gps_conversion.h [hunk 1]
```diff
@@ -22,19 +22,32 @@
 
 namespace msf_core {
 class GPSConversion {
- private:
-  msf_core::Quaternion ecef_ref_orientation_;
-  msf_core::Vector3 ecef_ref_point_;
```
Comment (tcies) [inline msf_core/include/msf_core/gps_conversion.h]: +1 thanks for moving this

File: msf_core/include/msf_core/gps_conversion.h [hunk 2]
```diff
@@ -22,19 +22,32 @@
...
- public:
+public:
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
   GPSConversion();
-  void InitReference(const double& latitude, const double& longitude,
+  // default copy constructor and assignment operator are OK
```
Comment (tcies) [inline msf_core/include/msf_core/gps_conversion.h]: Sentencify: Start with a cap, end with a period.

File: msf_core/include/msf_core/gps_conversion.h [hunk 3]
```diff
@@ -22,19 +22,32 @@
...
   msf_core::Vector3 WGS84ToECEF(const double& latitude,
                                 const double& longitude,
                                 const double& altitude) const;
+
   msf_core::Vector3 ECEFToENU(const msf_core::Vector3& ecef) const;
-  void AdjustReference(double z_corr);
+
+  msf_core::Vector3 WGS84ToENU(const double& latitude,
+                               const double& longitude,
+                               const double& altitude) const; // convenience
```
Comment (tcies) [inline msf_core/include/msf_core/gps_conversion.h]: Nit: No need for "convenience" comment.

Comment (tcies) [inline msf_core/include/msf_core/gps_conversion.h]: What? msf passes double by ref? This is inefficient. No need to refactor, but you could make your own function pass by copy.

File: msf_core/include/msf_core/gps_conversion.h [hunk 4]
```diff
@@ -22,19 +22,32 @@
...
+
+  void AdjustReference(const double& z_correction);
```
Comment (tcies) [inline msf_core/include/msf_core/gps_conversion.h]: +1 for const, -1 for ref

File: msf_core/src/lib/gps_conversion.cc [hunk 5]
```diff
@@ -21,14 +21,18 @@
 
 namespace msf_core {
 
-GPSConversion::GPSConversion() {
+GPSConversion::GPSConversion()
+{
```
Comment (tcies) [inline msf_core/src/lib/gps_conversion.cc]: This is not our convention. Please revert. Are you using the correct formatter? https://github.com/ethz-asl/programming_guidelines/wiki/Code-formatter

File: msf_core/src/lib/gps_conversion.cc [hunk 6]
```diff
@@ -21,14 +21,18 @@
...
   ecef_ref_orientation_.setIdentity();
 }
 
+GPSConversion::~GPSConversion() {}
+
 void GPSConversion::InitReference(const double & latitude,
                                   const double & longitude,
-                                  const double & altitude) {
+                                  const double & altitude)
+{
```
Comment (tcies) [inline msf_core/src/lib/gps_conversion.cc]: same

File: msf_core/src/lib/gps_conversion.cc [hunk 7]
```diff
@@ -51,9 +55,10 @@ void GPSConversion::InitReference(const double & latitude,
   ecef_ref_point_ = WGS84ToECEF(latitude, longitude, altitude);
 }
 
-msf_core::Vector3 GPSConversion::WGS84ToECEF(const double & latitude,
-                                             const double & longitude,
-                                             const double & altitude) const {
+msf_core::Vector3 GPSConversion::WGS84ToECEF(const double& latitude,
+                                             const double& longitude,
+                                             const double& altitude) const
+{
```
Comment (tcies) [inline msf_core/src/lib/gps_conversion.cc]: same

File: msf_core/src/lib/gps_conversion.cc [hunk 8]
```diff
@@ -72,8 +77,8 @@ msf_core::Vector3 GPSConversion::WGS84ToECEF(const double & latitude,
   return ecef;
 }
 
-msf_core::Vector3 GPSConversion::ECEFToENU(
-    const msf_core::Vector3 & ecef) const {
+msf_core::Vector3 GPSConversion::ECEFToENU(const msf_core::Vector3& ecef) const
+{
```
Comment (tcies) [inline msf_core/src/lib/gps_conversion.cc]: same

File: msf_core/src/lib/gps_conversion.cc [hunk 9]
```diff
@@ -81,10 +86,19 @@ msf_core::Vector3 GPSConversion::ECEFToENU(
...
 }
 
-void GPSConversion::AdjustReference(double z_corr) {
-  MSF_WARN_STREAM("z-ref old: "<<ecef_ref_point_(2));
-  ecef_ref_point_(2) += z_corr;
-  MSF_WARN_STREAM("z-ref new: "<<ecef_ref_point_(2));
+msf_core::Vector3 GPSConversion::WGS84ToENU(const double& latitude,
+                                            const double& longitude,
+                                            const double& altitude) const
+{
```
Comment (tcies) [inline msf_core/src/lib/gps_conversion.cc]: same

File: msf_core/src/lib/gps_conversion.cc [hunk 10]
```diff
@@ -81,10 +86,19 @@ msf_core::Vector3 GPSConversion::ECEFToENU(
...
+    msf_core::Vector3 ecef = this->WGS84ToECEF(latitude, longitude, altitude);
+    return this->ECEFToENU(ecef);
+}
+
+void GPSConversion::AdjustReference(const double& z_correction)
+{
```
Comment (tcies) [inline msf_core/src/lib/gps_conversion.cc]: same

File: msf_core/src/lib/gps_conversion.cc [hunk 11]
```diff
@@ -81,10 +86,19 @@ msf_core::Vector3 GPSConversion::ECEFToENU(
...
 
-void GPSConversion::AdjustReference(double z_corr) {
-  MSF_WARN_STREAM("z-ref old: "<<ecef_ref_point_(2));
-  ecef_ref_point_(2) += z_corr;
-  MSF_WARN_STREAM("z-ref new: "<<ecef_ref_point_(2));
+msf_core::Vector3 GPSConversion::WGS84ToENU(const double& latitude,
+                                            const double& longitude,
+                                            const double& altitude) const
+{
+    msf_core::Vector3 ecef = this->WGS84ToECEF(latitude, longitude, altitude);
```
Comment (tcies) [inline msf_core/src/lib/gps_conversion.cc]: const

**Comment by tcies** [comment]
Some comments, otherwise looks good to me (scolgtm; = please address comments, but no need for a second review)

**Comment by enricgalceran** [comment]
Thanks @tcies! Incorporated comments and refactored to pass POD types by reference on all functions in msf_core/gps_conversion.

Merging onto master.

File: msf_core/include/msf_core/gps_conversion.h [hunk 12]
```diff
@@ -22,19 +22,34 @@
...
+
   GPSConversion();
-  void InitReference(const double& latitude, const double& longitude,
-                     const double& altitude);
-  msf_core::Vector3 WGS84ToECEF(const double& latitude,
-                                const double& longitude,
-                                const double& altitude) const;
+  // Default copy constructor and assignment operator are OK.
+
+  ~GPSConversion();
```
Comment (markusachtelik) [inline msf_core/include/msf_core/gps_conversion.h]: can you please reformat class / method names according to this:
https://github.com/ethz-asl/programming_guidelines/wiki/Cpp-Coding-Style-Guidelines#10-abbreviations-and-acronyms-must-not-be-uppercase-when-used-as-name-4

I think it's still not too late ;)

**Pull request merged**